pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,344 @@
+	}
+
+	// create a request - mock the API call
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "rbac-permissions-operator",
+			Namespace: "default",
+		},
+	}
+
+	// make a new clusterRoleBinding by passing in clusterRoleName and groupName
+	// it needs to match our mockClusterRoleName
+	// newClusterRoleBinding := newClusterRoleBinding("exampleClusterRoleName", "exampleGroupName")
+
+	// if newClusterRoleBinding != mockClusterRoleBinding() {
+	// 	t.Errorf("got: %s, want: %s", newClusterRoleBinding, mockClusterRoleBinding())
+	// }
+	// get empty ClusterRoleList and give it a namespace
+
+	// process the request
+	res, err := reconciler.Reconcile(request)
+	if res.Requeue {
+		t.Error("Didn't expect to requeue")
+	}
+	if err != nil {
+		t.Errorf("Got an unexpected error: %s", err)
+	}
+}
+
+func TestPopulateClusterRoleBindingNames(t *testing.T) {
+	ctx := context.TODO()
+	reconciler := newTestReconciler()
+	defer reset(ctx, reconciler)
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+
+	//Add api to scheme
+	if err := routev1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add route scheme: (%v)", err)
+	}
+
+	list := &rbacv1.ClusterRoleBindingList{}
+	opts := realclient.ListOptions{Namespace: ""}
+
+	// create clusterRoleList{}
+	err := reconciler.client.List(ctx, &opts, list)
+	if err != nil {
+		t.Errorf("Couldn't get clusterRoleList for test: %s", err)
+	}
+
+	clusterRoleBindingName := []string{"1", "2", "3"}
+
+	tmpList := populateClusterRoleBindingNames(clusterRoleBindingName, list)
+	if tmpList != nil {
+		t.Errorf("this should be an empty list but: %s", tmpList)
+	}
+}
+
+func TestBuildClusterRoleBindingCrList(t *testing.T) {
+
+	buildList := buildClusterRoleBindingCRList(mockGroupPermissionSpec())
+
+	result := []string{"exampleClusterRoleName-exampleGroupName", "exampleClusterRoleNameTwo-exampleGroupName"}
+
+	if len(buildList) != len(result) {
+		t.Errorf("the length does not match")
+	}
+	for i, v := range result {
+		if v != buildList[i] {
+			t.Errorf("got %s, want %s", buildList, result)
+		}
+	}
+}
+
+func TestUpdateCondition(t *testing.T) {
+	time := metav1.Now()
+
+	buildCondition := updateCondition(mockGroupPermissionStatus(time), "testMessage", "testClusterRoleName", false, "testState")
+
+	testMap := make(map[int]managedv1alpha1.Condition)
+	initConOne := managedv1alpha1.Condition{
+		ClusterRoleName: "exampleClusterRoleName",
+		Message:         "exampleMessage",
+		Status:          true,
+		State:           "exampleState",
+	}
+	initConTwo := managedv1alpha1.Condition{
+		ClusterRoleName: "testClusterRoleName",
+		Message:         "testMessage",
+		Status:          false,
+		State:           "testState",
+	}
+
+	testMap[0] = initConOne
+	testMap[1] = initConTwo
+
+	for i, condition := range testMap {
+		if !(testCondition(condition, buildCondition.Status.Conditions[i])) {
```

Comment:
I'm not sure I like this approach to testing the conditions. I think a better test of conditions would be to perform some action on a GroupPermission object that would make the reconcile loop update the condition, and ensure that condition is what you expected.